test_unstructured/partition/test_new_email.py

```diff
@@ -0,0 +1,592 @@
+"""Test suite for `unstructured.partition.email` module."""
+
+from __future__ import annotations
+
+import io
+import tempfile
+from email.message import EmailMessage
+from typing import Any
+
+import pytest
+
+from test_unstructured.unit_utils import (
+    FixtureRequest,
+    Mock,
+    assert_round_trips_through_JSON,
+    example_doc_path,
+    function_mock,
+)
+from unstructured.chunking.title import chunk_by_title
+from unstructured.documents.elements import (
+    CompositeElement,
+    ListItem,
+    NarrativeText,
+    Table,
+    TableChunk,
+    Text,
+    Title,
+)
+from unstructured.partition.new_email import EmailPartitioningContext, partition_email
+
+EXPECTED_OUTPUT = [
+    NarrativeText(text="This is a test email to use for unit tests."),
+    Title(text="Important points:"),
+    ListItem(text="Roses are red"),
+    ListItem(text="Violets are blue"),
+]
+
+
+def test_partition_email_from_filename_can_partition_an_RFC_822_email():
+    assert partition_email(example_doc_path("eml/simple-rfc-822.eml")) == [
+        NarrativeText("This is an RFC 822 email message."),
+        NarrativeText(
+            "An RFC 822 message is characterized by its simple, text-based format, which includes"
+            ' a header and a body. The header contains structured fields such as "From", "To",'
+            ' "Date", and "Subject", each followed by a colon and the corresponding information.'
+            " The body follows the header, separated by a blank line, and contains the main"
+            " content of the email."
```

Comment:
Can you add an assert on the element metadata showing the additional email info that's mentioned here?

Reply:
We could I suppose, but this `elements == [Element("..."), ...]` testing is consistent with what we do for other email variants in the tests that follow it.

I purposely separated out the metadata tests so the failures are more specific and it's easier to tell what tests exercise what behavior. If we were doing strict black-box testing we would have a bunch of cases like N subtypes X M behaviors, but I don't see that paying off here. We happen to know that all email-specific metadata is applied on a particular line, not "subvariant" specific, So the one test (further down) is fine I think. Especially when you add the unit tests for each email-specific metadata value.